src/DataCollection.Shared/Binding Support/IdentifyController.cs

```diff
@@ -108,79 +104,97 @@ private void MapView_DoubleTapped(object sender, GeoViewInputEventArgs e)
-                    }
+                    _canellationTokenSource.Cancel();
+                    _canellationTokenSource = new CancellationTokenSource();
+                }
 
-                    // get the tap location in screen units and geographic coordinates
-                    var tapScreenPoint = e.Position;
-                    _tappedLocation = e.Location;
+                IsIdentifyInProgress = true;
+                var mapView = (MapView)sender;
+                var target = Target;
 
-                    // set identify parameters
-                    var pixelTolerance = 10;
-                    var returnPopupsOnly = false;
-                    var maxResultCount = 5;
 
-                    IReadOnlyList<IdentifyLayerResult> layerResults = null;
-                    IReadOnlyList<IdentifyGraphicsOverlayResult> graphicsOverlayResults = null;
-                    if (target == null)
-                    {
-                        // An identify target is not specified, so identify all layers and overlays
-                        var identifyLayersTask = mapView.IdentifyLayersAsync(tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount);
-                        var identifyOverlaysTask = mapView.IdentifyGraphicsOverlaysAsync(tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount);
-                        await Task.WhenAll(identifyLayersTask, identifyOverlaysTask);
-                        layerResults = identifyLayersTask.Result;
-                        graphicsOverlayResults = identifyOverlaysTask.Result;
-                    }
-                    else if (target is Layer targetLayer && mapView.Map.OperationalLayers.Contains(target as Layer))
+                if (target is ILoadable loadable && loadable.LoadStatus == LoadStatus.NotLoaded)
+                {
+                    await loadable.LoadAsync();
+                }
+
+                // get the tap location in screen units and geographic coordinates
+                var tapScreenPoint = e.Position;
+                _tappedLocation = e.Location;
+
+                // set identify parameters
+                var pixelTolerance = 10;
+                var returnPopupsOnly = false;
+                var maxResultCount = 5;
+
+                IReadOnlyList<IdentifyLayerResult> layerResults = null;
+                IReadOnlyList<IdentifyGraphicsOverlayResult> graphicsOverlayResults = null;
+                CancellationTokenSource _currentSource = _canellationTokenSource;
+                if (target == null)
+                {
+                    // An identify target is not specified, so identify all layers and overlays
+                    var identifyLayersTask = mapView.IdentifyLayersAsync(tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount, _currentSource.Token);
+                    var identifyOverlaysTask = mapView.IdentifyGraphicsOverlaysAsync(tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount);
```

Comment:
I guess it's an API oversight that `IdentifyGraphicsOverlaysAsync` doesn't have a cancel parameter?

Reply:
Yeah, my understanding is it isn't cancelable because theoretically it should be instantaneous.